crates/common/src/bail_manager.rs

```diff
@@ -0,0 +1,47 @@
+use std::sync::LazyLock;
+
+use serde::{Deserialize, Serialize};
+use tokio::sync::watch;
+
+/// Represents the context in which the system may need to "bail out."
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub enum BailContext {
+    /// Bail out after a sync event.
+    SyncEvent,
+    /// Bail out when a new block is processed in FCM.
+    FcmNewBlock,
+    /// Bail out during block signing.
+    SignBlock,
+}
```

Comment:
Just do this based on strings so it's more straightforward to define and pass in and it's more neutral to the use case.